scipy/linalg/_decomp_cossin.py

```diff
@@ -80,6 +80,13 @@ def cossin(X, p=None, q=None, separate=False,
         (``m-q`` x ``m-q``) orthogonal/unitary matrices. If ``separate=True``,
         this contains the tuple of ``(V1H, V2H)``.
 
+    Notes
+    -----
+    The documentation is written assuming array arguments are of specified
+    "core" shapes. However, array argument(s) of this function may have additional
```

Comment:
I just wanted to double check, the decorator places this in the extended summary are you happy having it in a different location for this function?

Reply:
The extended summary was already quite long for this function, so I thought this would be more visible. It doesn't bother me, personally.

Another option is to move much of this extended summary in the notes to make the extended summary shorter. Then I might put the snippet up there.

Or TBH, maybe the snippet should go in the notes across the board?

Reply:
It doesn't bother me either (`linalg` seems to follow a pattern of slightly overgrown extended summaries) just wanted to double check.

I am happy to merge as in since we might want to look at documentation after everything is done anyway since functions that had existing batch support document this differently to those that use the decorator.